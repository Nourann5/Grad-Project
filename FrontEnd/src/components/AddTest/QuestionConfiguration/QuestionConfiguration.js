import React, { useEffect, useState } from 'react'
import styles from './QuestionConfiguration.module.css'
import { useFormContext } from 'react-hook-form';
import {ReactComponent as PlusIcon} from 'assets/icons/plus.svg'
import arabicAlphabetSrc from 'assets/imgs/arabicAlphabet.jpg'
import englishAlphabetSrc from 'assets/imgs/englishAlphabet.jpg'
import numberAlphabetSrc from 'assets/imgs/numberAlphaBet.jpg'
import romanAlphabetSrc from 'assets/imgs/romanAlpha.jpg'
import {ReactComponent as CheckedIcon} from 'assets/icons/checked.svg'
import {ReactComponent as DownladExcelIcon} from 'assets/icons/downladExcel.svg'
import * as XLSX from 'xlsx'
import {QUESTION_MODAL ,QUESTION_ANSWER_MODAL} from 'views/PortalPages/AddTest/AddTest'
import { Link } from 'react-router-dom';
import fileSrc from 'assets/files/questions-modal.xlsx'
function QuestionConfiguration({addQuestionByExcelFile}) {

    const { register ,setValue ,getValues} = useFormContext({
        defaultValues:{
            examConfiguration:{
                no_of_question:1,
                alpha_type:'number'
            }
        }
    });
    
    const [questionNumber , setQuestionNumber] =useState(getValues('examConfiguration.no_of_question'))

    // let changeExamData =(e,type)=>{
    //     let modififedExamData =[...examData]
    //     modififedExamData[questionIndex][type]=e.target.value;
    //     console.log(modififedExamData)
    //     setExamData(modififedExamData)
    // }
    let changeQuestionNumber= (e)=>{
        let modifiedData =  e.target.value>600? 600  
                            :e.target.value<1 ?1:e.target.value
        setQuestionNumber(modifiedData)
    }
    // let addQuestionByExcelFile =async (e)=>{
    //     const file = e.target.files[0];
    //     const data = await file.arrayBuffer();
    //     const workbook = XLSX.read(data);
    //     const workSheet = workbook.Sheets[workbook.SheetNames[0]]
    //     const jsonData = XLSX.utils.sheet_to_json(workSheet) 
    //     console.log(jsonData)
    //     addExcelFileQuestionsToExamData(jsonData)
    // }
    
    // let addExcelFileQuestionsToExamData = (excelData)=>{
    //     let modifiedExamData = [];
    //     console.log(modifiedExamData)
    //     excelData.forEach((question,questionIndex)=>{
    //         modifiedExamData.push(QUESTION_MODAL(questionIndex+1,49,question))
    //     })
    //     setExamData(modifiedExamData)
    // }
    useEffect(()=>{
        setQuestionNumber(getValues('examConfiguration.no_of_question'))
    },[getValues('examConfiguration.no_of_question')])
  return (
    <>
        
        <div className={styles['add-test__configuration-field-cont']}>
            <div className={`${styles['add-test__configuration-label-cont']} d-flex flex-column align-items-start`}>
                <label className={styles["add-test__config-label"]}>Question Number</label>
                <div className='d-flex w-100 mt-3'>
                    <input 
                        type="number" 
                        min='1' 
                        step='1' 
                        className={`${styles['add-test__configuration-input']} form-control`} 
                        placeholder='please add Exam Time'
                        value={questionNumber}
                        onChange={(e)=>{changeQuestionNumber(e)}}
                    />
                    <button 
                    className={`${styles['question-number__add-button']}
                    ${getValues('examConfiguration.no_of_question')===questionNumber ?styles['question-number__add-button--not-allowed']:''}`}
                    onClick={()=>{setValue('examConfiguration.no_of_question',questionNumber)}}
                    disabled={getValues('examConfiguration.no_of_question')===questionNumber}
                    >
                        <PlusIcon className={styles['question-number__add-button-icon']}/>
                    </button>
                </div>
            </div>
        </div>

        <div className={styles['question-control__answer-type-wrapper']}>
            <div className='d-flex align-items-center'>
                <label className={styles['question-control__answer-type-lable']}>Question Alpha</label>
            </div>
            <div className={`${styles['question-control__screens-type-cont']}`}>
                <div className={styles['question-control__screen-type']}>
                    <input 
                            type='radio' 
                            value='number'
                            className={styles['question-control__screen-type-input']}
                            id='answernumber'
                            {...register('examConfiguration.alpha_type')}
                            defaultChecked={getValues('examConfiguration.alpha_type')=='number'}
                        />

                    <label htmlFor='answernumber' className={styles['question-control__screen-type-label']}>
                        <img src={numberAlphabetSrc} alt='number select' className={styles['question-control__screen-type-img']}/>
                        <div className={`${styles['question-control__screen-type--selected']}
                        ${getValues('examConfiguration.alpha_type')=='number'?'':'d-none'}`}>
                                <CheckedIcon className={styles['question__checked-icon']}/>
                        </div>
                    </label>
                </div>
                
                <div className={styles['question-control__screen-type']}>
                    <input 
                        type='radio' 
                        value='english' 
                        id='answerEnglish'
                        className={styles['question-control__screen-type-input']}
                        {...register('examConfiguration.alpha_type')}
                        defaultChecked={getValues('examConfiguration.alpha_type')=='english'}
                    />
                    <label htmlFor='answerEnglish' className={styles['question-control__screen-type-label']}>
                        <img src={englishAlphabetSrc} alt='english select' className={styles['question-control__screen-type-img']}/>
                        <div className={`${styles['question-control__screen-type--selected']}
                            ${getValues('examConfiguration.alpha_type')=='english'?'':'d-none'}`}>
                                <CheckedIcon className={styles['question__checked-icon']}/>
                        </div>
                    </label>
                </div>

                {/* <div className={styles['question-control__screen-type']}>
                        <input 
                        type='radio' 
                        value='arabic' 
                        id='answerarabic'
                        className={styles['question-control__screen-type-input']}
                        {...register('examConfiguration.alpha_type')}
                    defaultChecked={getValues('examConfiguration.alpha_type')=='arabic'}
                    />
                    <label htmlFor='answerarabic' className={styles['question-control__screen-type-label']}>
                        <img src={arabicAlphabetSrc} alt='arabic' className={styles['question-control__screen-type-img']}/>
                        <div className={`${styles['question-control__screen-type--selected']}
                            ${getValues('examConfiguration.alpha_type')=='arabic'?'':'d-none'}`}>
                                <CheckedIcon className={styles['question__checked-icon']}/>
                        </div>
                    </label>
                </div> */}

                <div className={styles['question-control__screen-type']}>
                    <input 
                        type='radio' 
                        value='roman' 
                        id='answerRoman'
                        className={styles['question-control__screen-type-input']}
                        {...register('examConfiguration.alpha_type')}
                        defaultChecked={getValues('examConfiguration.alpha_type')=='roman'}
                    />
                    <label htmlFor='answerRoman' className={styles['question-control__screen-type-label']}>
                        <img src={romanAlphabetSrc} alt='roman' className={styles['question-control__screen-type-img']}/>
                        <div className={`${styles['question-control__screen-type--selected']}
                            ${getValues('examConfiguration.alpha_type')=='roman'?'':'d-none'}`}>
                                <CheckedIcon className={styles['question__checked-icon']}/>
                        </div>
                    </label>

                </div>
            </div>
        </div>
        
        <div className={styles['add-test__configuration-field-cont']}>
            <div className={`${styles['add-test__configuration-label-cont']} d-flex flex-column align-items-start`}>
                <div className='d-flex align-items-center w-100'>

                    <label className={styles["add-test__config-label"]}>Add Question By Excel File</label>
                    <a download href={fileSrc} className={styles['add-test__config-download-button']}>
                        <DownladExcelIcon className={styles['add-test__config-download-icon']}/>
                    </a>
                </div>
                <div className='d-flex w-100 mt-3'>
                    <div className={styles['add-test__configuration-custom-input-wrapper']}>
                        <label htmlFor='add-test__configuration-input' className={styles['add-test__configuration-custom-input-label']}>
                            Upload File
                        </label>
                        <input 
                            type="file" 
                            className={`${styles['add-test__configuration-input--file']} form-control`} 
                            placeholder='please add Exam Time'
                            accept='.xlsx,.xls'
                            id='add-test__configuration-input'
                            onChange={(e)=>{addQuestionByExcelFile(e)}}
                        />
                    </div>
                </div>
            </div>
        </div>
        
    </>
  )
}

export default QuestionConfiguration