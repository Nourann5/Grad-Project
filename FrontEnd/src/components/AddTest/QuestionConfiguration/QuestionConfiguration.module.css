
.add-test__configuration-accordion-body{
    padding: 20px 40px;
    border-top: 1px solid #00000012;
}
.add-test__configuration-accordion-item{
    /* margin-bottom: 30px; */
    border: 0;
    border-top: 1px solid #00000012;
    border-bottom: 1px solid #00000012;
}
.add-test__configuration-label-cont{
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.add-test__config-label{
    font-family: 'PoppinsSemiBold';
}
.add-test__configuration-field-cont{
    margin-top: 20px;
    margin-bottom: 20px;
}
.add-test__configuration-input{
    border-radius: 0!important;
    border-right-color: transparent!important;
}
.add-test__configuration-input:focus-visible ,.add-test__configuration-input:focus{
    border-color: var(--main-color);
}
.add-test__configuration-input--file{
    display: none;
    border-radius: 0!important;
}
.add-test__configuration-input--file:focus-visible ,.add-test__configuration-input--file:focus{
    border-color: var(--main-color)!important;
}
.add-test__configuration-custom-input-wrapper{
    width: 100%;
}
.add-test__config-download-button{
    margin-left: auto;
}
.add-test__config-download-icon{
    width: 40px;
    height: 40px;
}
.add-test__configuration-custom-input-label{
    width: 100%;
    height: 50px;
    background-color: var(--main-color);
    color: white;
    display: flex;
    align-items: center;
    justify-content:center;
    cursor: pointer;
    transition: var(--transition);
}
.add-test__configuration-custom-input-label:hover{
    opacity: .9;
}
.question-number__add-button{
    background-color: transparent;
    border: 2px solid var(--main-color)!important;
    width: 40px;
}
.question-number__add-button-icon{
    width: 15px;
    height: 15px;
    color: var(--main-color);
}
.question-number__add-button--not-allowed{
    cursor: not-allowed!important;
    pointer-events: initial!important;
    opacity: .7;
}



.question-control__screens-type-cont{
    display: grid;
    grid-template-columns: repeat(3,1fr);
    grid-gap: 7px;
    margin-top: 10px;
}

.question-control__screen-type-input{
    display: none;
}
.question-control__screen-type-img{
    padding: 5px;
    width: 100%;
    height: 80px;
    border: 2px solid rgb(0 0 0 / 50%);
    border-radius: 5px;
}
.question-control__screens-type-cont .question-control__screen-type-img{
    height: 50px;
}
.question-control__screen-type-label{
    cursor: pointer;
    position: relative!important;
}
.question-control__screen-type--selected{
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    display: flex;
    align-items: center;
    justify-content: center;
}
.question-control__screen-type--selected::before{
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: white;
    opacity: .6;
}
.question__checked-icon{
    z-index: 1;
    color: green;
    width: 30px;
    height: 30px;
}

@media (max-width:992px){
    .question-control__screen-type{
        margin: auto;
    }
}