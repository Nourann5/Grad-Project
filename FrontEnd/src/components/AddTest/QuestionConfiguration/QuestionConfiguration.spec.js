import {render , renderHook, screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import { FormProvider, useForm } from 'react-hook-form'
import QuestionConfiguration from './QuestionConfiguration'

const methods = renderHook(()=> useForm())

describe('Test the Add Test ', () => { 
    it('QuestionConfiguration',()=>{
        const TestInput = () => {
            const methods = useForm();
            return (
                <BrowserRouter>
                    <FormProvider {...methods}>
                        <QuestionConfiguration />
                    </FormProvider>
                </BrowserRouter>
            );
        };
        const tree =  render(<TestInput/>)
        expect(tree).toMatchSnapshot()
    })
})