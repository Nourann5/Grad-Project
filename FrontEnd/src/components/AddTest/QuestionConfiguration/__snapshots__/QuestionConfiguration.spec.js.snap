// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Add Test  QuestionConfiguration 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="add-test__configuration-field-cont"
      >
        <div
          class="add-test__configuration-label-cont d-flex flex-column align-items-start"
        >
          <label
            class="add-test__config-label"
          >
            Question Number
          </label>
          <div
            class="d-flex w-100 mt-3"
          >
            <input
              class="add-test__configuration-input form-control"
              min="1"
              placeholder="please add Exam Time"
              step="1"
              type="number"
              value=""
            />
            <button
              class="question-number__add-button
                    question-number__add-button--not-allowed"
              disabled=""
            >
              <svg
                class="question-number__add-button-icon"
              >
                plus.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="question-control__answer-type-wrapper"
      >
        <div
          class="d-flex align-items-center"
        >
          <label
            class="question-control__answer-type-lable"
          >
            Question Alpha
          </label>
        </div>
        <div
          class="question-control__screens-type-cont"
        >
          <div
            class="question-control__screen-type"
          >
            <input
              class="question-control__screen-type-input"
              id="answernumber"
              name="examConfiguration.alpha_type"
              type="radio"
              value="number"
            />
            <label
              class="question-control__screen-type-label"
              for="answernumber"
            >
              <img
                alt="number select"
                class="question-control__screen-type-img"
                src="numberAlphaBet.jpg"
              />
              <div
                class="question-control__screen-type--selected
                        d-none"
              >
                <svg
                  class="question__checked-icon"
                >
                  checked.svg
                </svg>
              </div>
            </label>
          </div>
          <div
            class="question-control__screen-type"
          >
            <input
              class="question-control__screen-type-input"
              id="answerEnglish"
              name="examConfiguration.alpha_type"
              type="radio"
              value="english"
            />
            <label
              class="question-control__screen-type-label"
              for="answerEnglish"
            >
              <img
                alt="english select"
                class="question-control__screen-type-img"
                src="englishAlphabet.jpg"
              />
              <div
                class="question-control__screen-type--selected
                            d-none"
              >
                <svg
                  class="question__checked-icon"
                >
                  checked.svg
                </svg>
              </div>
            </label>
          </div>
          <div
            class="question-control__screen-type"
          >
            <input
              class="question-control__screen-type-input"
              id="answerRoman"
              name="examConfiguration.alpha_type"
              type="radio"
              value="roman"
            />
            <label
              class="question-control__screen-type-label"
              for="answerRoman"
            >
              <img
                alt="roman"
                class="question-control__screen-type-img"
                src="romanAlpha.jpg"
              />
              <div
                class="question-control__screen-type--selected
                            d-none"
              >
                <svg
                  class="question__checked-icon"
                >
                  checked.svg
                </svg>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="add-test__configuration-field-cont"
      >
        <div
          class="add-test__configuration-label-cont d-flex flex-column align-items-start"
        >
          <div
            class="d-flex align-items-center w-100"
          >
            <label
              class="add-test__config-label"
            >
              Add Question By Excel File
            </label>
            <a
              class="add-test__config-download-button"
              download=""
              href="questions-modal.xlsx"
            >
              <svg
                class="add-test__config-download-icon"
              >
                downladExcel.svg
              </svg>
            </a>
          </div>
          <div
            class="d-flex w-100 mt-3"
          >
            <div
              class="add-test__configuration-custom-input-wrapper"
            >
              <label
                class="add-test__configuration-custom-input-label"
                for="add-test__configuration-input"
              >
                Upload File
              </label>
              <input
                accept=".xlsx,.xls"
                class="add-test__configuration-input--file form-control"
                id="add-test__configuration-input"
                placeholder="please add Exam Time"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="add-test__configuration-field-cont"
    >
      <div
        class="add-test__configuration-label-cont d-flex flex-column align-items-start"
      >
        <label
          class="add-test__config-label"
        >
          Question Number
        </label>
        <div
          class="d-flex w-100 mt-3"
        >
          <input
            class="add-test__configuration-input form-control"
            min="1"
            placeholder="please add Exam Time"
            step="1"
            type="number"
            value=""
          />
          <button
            class="question-number__add-button
                    question-number__add-button--not-allowed"
            disabled=""
          >
            <svg
              class="question-number__add-button-icon"
            >
              plus.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="question-control__answer-type-wrapper"
    >
      <div
        class="d-flex align-items-center"
      >
        <label
          class="question-control__answer-type-lable"
        >
          Question Alpha
        </label>
      </div>
      <div
        class="question-control__screens-type-cont"
      >
        <div
          class="question-control__screen-type"
        >
          <input
            class="question-control__screen-type-input"
            id="answernumber"
            name="examConfiguration.alpha_type"
            type="radio"
            value="number"
          />
          <label
            class="question-control__screen-type-label"
            for="answernumber"
          >
            <img
              alt="number select"
              class="question-control__screen-type-img"
              src="numberAlphaBet.jpg"
            />
            <div
              class="question-control__screen-type--selected
                        d-none"
            >
              <svg
                class="question__checked-icon"
              >
                checked.svg
              </svg>
            </div>
          </label>
        </div>
        <div
          class="question-control__screen-type"
        >
          <input
            class="question-control__screen-type-input"
            id="answerEnglish"
            name="examConfiguration.alpha_type"
            type="radio"
            value="english"
          />
          <label
            class="question-control__screen-type-label"
            for="answerEnglish"
          >
            <img
              alt="english select"
              class="question-control__screen-type-img"
              src="englishAlphabet.jpg"
            />
            <div
              class="question-control__screen-type--selected
                            d-none"
            >
              <svg
                class="question__checked-icon"
              >
                checked.svg
              </svg>
            </div>
          </label>
        </div>
        <div
          class="question-control__screen-type"
        >
          <input
            class="question-control__screen-type-input"
            id="answerRoman"
            name="examConfiguration.alpha_type"
            type="radio"
            value="roman"
          />
          <label
            class="question-control__screen-type-label"
            for="answerRoman"
          >
            <img
              alt="roman"
              class="question-control__screen-type-img"
              src="romanAlpha.jpg"
            />
            <div
              class="question-control__screen-type--selected
                            d-none"
            >
              <svg
                class="question__checked-icon"
              >
                checked.svg
              </svg>
            </div>
          </label>
        </div>
      </div>
    </div>
    <div
      class="add-test__configuration-field-cont"
    >
      <div
        class="add-test__configuration-label-cont d-flex flex-column align-items-start"
      >
        <div
          class="d-flex align-items-center w-100"
        >
          <label
            class="add-test__config-label"
          >
            Add Question By Excel File
          </label>
          <a
            class="add-test__config-download-button"
            download=""
            href="questions-modal.xlsx"
          >
            <svg
              class="add-test__config-download-icon"
            >
              downladExcel.svg
            </svg>
          </a>
        </div>
        <div
          class="d-flex w-100 mt-3"
        >
          <div
            class="add-test__configuration-custom-input-wrapper"
          >
            <label
              class="add-test__configuration-custom-input-label"
              for="add-test__configuration-input"
            >
              Upload File
            </label>
            <input
              accept=".xlsx,.xls"
              class="add-test__configuration-input--file form-control"
              id="add-test__configuration-input"
              placeholder="please add Exam Time"
              type="file"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
